### PR TITLE
Tab Animation Sample Video

### LINKED ISSUE #31392: TabBar and TabBarView documentation could be clearer
- [x] The documentation for [TabBar](https://docs.flutter.io/flutter/material/TabBar-class.html) and [TabBarView](https://docs.flutter.io/flutter/material/TabBarView-class.html) have sample implementations in [TabBarController](https://docs.flutter.io/flutter/material/TabController-class.html) and [DefaultTabController](https://docs.flutter.io/flutter/material/DefaultTabController-class.html), but this isn't clear from the get-go
- [ ] There could also be screenshots or a short gif of what a TabBar/TabBarView combination visually looks like to help first-time users of the widgets.
- [x] [DefaultTabController.length](https://docs.flutter.io/flutter/material/DefaultTabController/length.html), [TabController.length](https://docs.flutter.io/flutter/material/TabController/length.html) and [TabBarView.children](https://docs.flutter.io/flutter/material/TabBarView/children.html) do not specify required number of tabs, children and the controller lengths have to match, while [TabBar.tabs](https://docs.flutter.io/flutter/material/TabBar/tabs.html) does, which can be confusing.

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

Adds a video of what a TabBar/TabBarView combination looks like to the documentation: https://flutter.github.io/assets-for-api-docs/assets/material/tabs.mp4

Created with https://github.com/flutter/assets-for-api-docs/pull/72

## Related Issues

Fixes https://github.com/flutter/flutter/issues/31392

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 1]
```diff
@@ -32,8 +34,8 @@ import 'constants.dart';
 ///
 /// class _MyTabbedPageState extends State<MyTabbedPage> with SingleTickerProviderStateMixin {
 ///   final List<Tab> myTabs = <Tab>[
-///     Tab(text: 'LEFT'),
-///     Tab(text: 'RIGHT'),
+///     Tab(text: 'left'),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/tab_controller.dart]: The convention per the Material spec is call caps for tab titles

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 2]
```diff
@@ -71,13 +78,15 @@ import 'constants.dart';
 /// ```
 /// {@end-tool}
 class TabController extends ChangeNotifier {
-  /// Creates an object that manages the state required by [TabBar] and a [TabBarView].
+  /// Creates an object that manages the state required by [TabBar] and a
+  /// [TabBarView].
   ///
-  /// The [length] must not be null or negative. Typically its a value greater than one, i.e.
-  /// typically there are two or more tabs.
+  /// The [length] must not be null or negative. Typically its a value greater
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/tab_controller.dart]: its => it's

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 3]
```diff
@@ -219,11 +229,13 @@ class _TabControllerScope extends InheritedWidget {
...
 /// widgets.
 ///
+/// {@animation 700 540 https://flutter.github.io/assets-for-api-docs/assets/material/tabs.mp4}
+///
 /// ```dart
 /// class MyDemo extends StatelessWidget {
 ///   final List<Tab> myTabs = <Tab>[
-///     Tab(text: 'LEFT'),
-///     Tab(text: 'RIGHT'),
+///     Tab(text: 'left'),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/tab_controller.dart]: LEFT and RIGHT is OK

File: packages/flutter/lib/src/material/tabs.dart [hunk 4]
```diff
@@ -1089,7 +1098,8 @@ class TabBarView extends StatefulWidget {
   /// will be used.
   final TabController controller;
 
-  /// One widget per tab.
+  /// One widget per tab. Its length must match the length of the [TabBar.tabs]
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/tabs.dart]: Insert a blank line after "One widget per tab". The format for docs should be:
```
/// Brief but surprisingly comprehensive summary statement (one sentence).
///
/// Additional explanation.
```

File: packages/flutter/lib/src/material/tab_controller.dart (hunk 1 above)
Comment (shihaohong) [reply packages/flutter/lib/src/material/tab_controller.dart]: I reverted it to all caps

File: packages/flutter/lib/src/material/tabs.dart (hunk 4 above)
Comment (shihaohong) [reply packages/flutter/lib/src/material/tabs.dart]: Sounds good!

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**